src/software/simulated_tests/simulated_test_fixture.h

```diff
@@ -2,40 +2,165 @@
 
 #include <gtest/gtest.h>
 
-#include "software/ai/ai_wrapper.h"
-#include "software/backend/simulator_backend.h"
+#include "software/ai/ai.h"
 #include "software/gui/visualizer/visualizer_wrapper.h"
-#include "software/simulated_tests/validation/world_state_validator.h"
+#include "software/sensor_fusion/sensor_fusion.h"
+#include "software/simulated_tests/validation/continuous_function_validator.h"
+#include "software/simulated_tests/validation/function_validator.h"
+#include "software/simulation/simulator.h"
 
 /**
- * This is a test fixture designed to make it easy to write integration tests with the
- * SimulatorBackend. It uses the SimulatorBackend to simulate and publish the World, and
- * the WorldStateValidator to make assertions about the world. This allows us to easily
- * write tests for the AI's behaviour.
+ * This is a test fixture designed to make it easy to write integration tests. It provides
+ * an easy interface to set up robots on the field, and then validate how the world
+ * changes over time during simulation. This allows us to easily write tests for
+ * the AI's behaviour.
  */
 class SimulatedTest : public ::testing::Test
 {
+   public:
+    explicit SimulatedTest();
+
    protected:
+    void SetUp() override;
+
     /**
-     * The setup function that will run before each test case
+     * This function enables the Visualizer while a test is running, so that the test can
+     * be debugged Visually. Simply call this function at the start of the test(s) you
+     * want to show in the Visualizer.
      */
-    void SetUp() override;
+    void enableVisualizer();
 
     /**
-     * The teardown function that will run after each test case
+     * Starts the simulation using the current state of the simulator, and runs
+     * the given ValidationFunctions on each new state of the World. The test
+     * will succeed if
+     * - There are no validation_functions and all continuous_validation_functions
+     *   pass for the duration of the test. In this case the test will run for the
+     *   duration of the timeout.
+     * - There is at least one validation_function, and all validation functions
+     *   continuous and non-continuous) pass. In this case the test will run until
+     *   all validation_functions have passed or the timeout is exceeded, whichever
```

Comment:
After offline discussion, we decided to rename the two groups of functions to `terminating_validation_functions` and `non_terminating_validation_functions` to help clarify the distinction between the two.

Reply:
Ok I renamed
* `FunctionValidator` to `TerminatingFunctionValidator`
* `ContinuousFunctionValidator` to `NonTerminatingFunctionValidator`
* Renamed all variables like `validation_functions` and `continuous_validation_functions` similarly where appropriate, and updated all comments